scripts/Connector/Connector_JST/conn_jst_XA_horizontal.py

```diff
@@ -0,0 +1,272 @@
+        orientation=orientation_str, man=manufacturer,
+        entry=configuration['entry_direction'][orientation])
+    if hook:
+        tags += ' hook'
+    kicad_mod.setTags(tags)
+
+
+    # create Silkscreen
+    tmp_x1=x_min+body_back_protrusion_width+configuration['silk_fab_offset']
+    tmp_x2=x_max-body_back_protrusion_width-configuration['silk_fab_offset']
+    pad_silk_offset = configuration['silk_pad_clearance'] + configuration['silk_line_width']/2
+    poly_silk_outline= [
+                    {'x':-pad_size[0]/2.0-pad_silk_offset, 'y':silk_y_main_min},
+                    {'x':tmp_x1, 'y':silk_y_main_min},
+                    {'x':tmp_x1, 'y':silk_y_min},
+                    {'x':silk_x_min, 'y':silk_y_min},
+                    {'x':silk_x_min, 'y':silk_y_max},
+                    {'x':silk_x_max, 'y':silk_y_max},
+                    {'x':silk_x_max, 'y':silk_y_min},
+                    {'x':tmp_x2, 'y':silk_y_min},
+                    {'x':tmp_x2, 'y':silk_y_main_min},
+                    {'x':(pincount-1)*pitch+pad_size[0]/2.0+pad_silk_offset, 'y':silk_y_main_min}
+    ]
+    kicad_mod.append(PolygoneLine(polygone=poly_silk_outline, layer='F.SilkS', width=configuration['silk_line_width']))
+
+    if configuration['allow_silk_below_part'] == 'tht' or configuration['allow_silk_below_part'] == 'both':
+        if hook:
+            poly_big_cutout=[{'x':x_min+1, 'y':silk_y_max}
+                                  ,{'x':x_min+1, 'y':silk_y_max - 2.3}
+                                  ,{'x':(A/2)-1.8, 'y':silk_y_max - 2.3}]
+            kicad_mod.append(PolygoneLine(polygone=poly_big_cutout, layer='F.SilkS', width=configuration['silk_line_width']))
+            poly_big_cutout=[{'x':(A/2)+1.8, 'y':silk_y_max-2.3}
+                                  ,{'x':x_max-1, 'y':silk_y_max - 2.3}
+                                  ,{'x':x_max-1, 'y':silk_y_max}]
+        else:
+            poly_big_cutout=[{'x':x_min+1, 'y':silk_y_max}
+                                  ,{'x':x_min+1, 'y':silk_y_max - 2.3}
+                                  ,{'x':x_max-1, 'y':silk_y_max - 2.3}
+                                  ,{'x':x_max-1, 'y':silk_y_max}]
+        kicad_mod.append(PolygoneLine(polygone=poly_big_cutout, layer='F.SilkS', width=configuration['silk_line_width']))
+
+        kicad_mod.append(Line(start=[silk_x_min, silk_y_main_min], end=[tmp_x1, silk_y_main_min], layer='F.SilkS', width=configuration['silk_line_width']))
+        kicad_mod.append(Line(start=[silk_x_max, silk_y_main_min], end=[tmp_x2, silk_y_main_min], layer='F.SilkS', width=configuration['silk_line_width']))
+
+
+        #locking "dimple", centered
+        dimple_w = 1.6
+        dimple_h = 1.8
+        dimple_y = y_max - 4.9  #middle of the dimple
+        kicad_mod.append(RectLine(start=[(A-dimple_w)/2, dimple_y + dimple_h/2], end=[(A+dimple_w)/2, dimple_y - dimple_h/2],
+            layer='F.SilkS', width=configuration['silk_line_width']))
+
```

Comment:
Ok, interesting point, but as you've shown there's at least 2 different "types" of underside ribbing that would need to be scripted depending on # of pins ? I'm happy to add this, but I'm out of time to go through all the CAD models again and figure out the "system"...

Reply:
i looked at them, and it seems to really be just the two types.
* 2 and 3 pin have just the paralell 1mm walls at the end, that's it.
* `>= 4` pins have the parralell walls at the end, and always the same center part.
(i haven't measured, but it looks like this center part is exactly the same width as the walls of a 2 pin connector).
  plus a wall at the front which connects the walls at the ends with the walls in the middle

i tried sketching them quickly (grey: courtyard, yellow: pads, red: mech hole, blue: silkscreen):
![2020-04-02-172923](https://user-images.githubusercontent.com/22578460/78267735-913aeb80-7507-11ea-9a56-7d6ba716f1e3.png)
![2020-04-02-173039](https://user-images.githubusercontent.com/22578460/78267810-add72380-7507-11ea-8649-c4f4a9155900.png)
and i must say, i'm actually not quite sure i like them. especially the 2 and 3 pin parts would be quite empty.

Reply:
hmm. I like your courtyard better, but I think showing the bottom of the connector is less intuitive than the top ? Other _Horizontal connectors also seem to match the idea of a "top view" better. I will un-close the back legs like you mentioned though.

Reply:
you're right, it makes more sense to have the top view on the silkscreen.
not having the connector physically here yet and just looking at the silkscreen it wasn't obvious from the start what i see there, though.
it might make sense to resemble the top of the connector a little more closely.